offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/components/product/MaintainProductsView.groovy

```diff
@@ -157,14 +163,38 @@ class MaintainProductsView extends VerticalLayout{
             createProductView.setVisible(false)
         })
 
+        copyProduct.addClickListener({
+            viewModel.productUpdate.emit(viewModel.selectedProduct)
+            maintenanceLayout.setVisible(false)
+            copyProductView.setVisible(true)
+        })
+
+        copyProductView.abortButton.addClickListener({
+            maintenanceLayout.setVisible(true)
+            copyProductView.setVisible(false)
+        })
+
         archiveProduct.addClickListener({
             controller.archiveProduct(viewModel.selectedProduct.productId)
         })
 
         viewModel.products.addPropertyChangeListener({
             productGrid.dataProvider.refreshAll()
         })
+        viewModel.addPropertyChangeListener({
+            checkProductSelected(viewModel)
```

Comment:
Why do you need the parameter? You don't use it and it is a class variable
```suggestion
            checkProductSelected()
```

Reply:
Done!